Move compiled method installation code from compile to install step

guillep opened the pull request:
The compile step tries to fill a method with metadata existing on installed methods.
Fetching this existing method and filling the meta-data should be done at the installation step and not the compilation step.

This is important for the bootstrap step, where no prior method exists, and no prior class to look for is there either.

guillep committed:
Move method property persistence to installation phase

MarcusDenker commented:
fails with which I think is not caused by this PR:

```
SystemNotification: Loading Compression-tonel.1.mcz
[547](https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-17469/1/pipeline-console/?start-byte=0&selected-node=26#log-547)
KeyNotFound: key #unzipped not found in MethodDictionary
[548](https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-17469/1/pipeline-console/?start-byte=0&selected-node=26#log-548)
MethodDictionary(Dictionary)>>errorKeyNotFound:
[549](https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-17469/1/pipeline-console/?start-byte=0&selected-node=26#log-549)
[self errorKeyNotFound: key] in MethodDictionary(Dictionary)>>at: in Block: [self errorKeyNotFound: key]
[550](https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-17469/1/pipeline-console/?start-byte=0&selected-node=26#log-550)
MethodDictionary>>at:ifAbsent:
[551](https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-17469/1/pipeline-console/?start-byte=0&selected-node=26#log-551)
MethodDictionary(Dictionary)>>at:
[552](https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-17469/1/pipeline-console/?start-byte=0&selected-node=26#log-552)
ByteArray class(Behavior)>>compiledMethodAt:
```

MarcusDenker commented:
Ah, this is Pharo12, I added an issue: https://github.com/pharo-project/pharo/issues/17472

MarcusDenker commented:
So the Problem

```KeyNotFound: key #unzipped not found in MethodDictionary"```

Is indeed caused by this PR

guillep commented:
I'll check it during the week

guillep committed:
Fix(recompilation): Make use of install instead of compile during recompilation from a class.
It is the compiler in charge of installing a method and maintain invariants such as annotations

guillep committed:
Do not announce when recompiling

jecisc commented:
Build passed but there are a lot a failing tests. But they are weird because most of them are about dependencies but I don't see how this change bring those dependencies.

guillep commented:
Oh but now that this passed I can take a look in-image :)

guillep committed:
Migrate properties in reflectivity

guillep commented:
> Build passed but there are a lot a failing tests. But they are weird because most of them are about dependencies but I don't see how this change bring those dependencies.

To anwer shortly, first, this PR moves method installation responsibilities to the `install` method.
The issue is that many places assumed that method compilation did things that arguably should not (like marking methods as extensions) and then code installation is duplicated in many places.

I just catched one more, probably there are more to come.

guillep commented:
Looks ok now! 
<img width="986" alt="imagen" src="https://github.com/user-attachments/assets/4add3ec9-f28f-42ce-bfc3-69be8c53a503" />
